Update comments of two functions.

bangerth opened the pull request:
These functions were not individually documented. Fix this and make
clear what they do.

drwells commented on include/deal.II/hp/dof_handler.h:
```diff
@@ -928,14 +928,23 @@ namespace hp
     /**
-     * Functions that will be triggered through signals whenever the
-     * triangulation is modified.
+     * A functions that will be triggered through a triangulation
```
Should be `A function`

jppelteret commented on include/deal.II/hp/dof_handler.h:
```diff
@@ -928,14 +928,23 @@ namespace hp
     pre_refinement_action();
+
+    /**
+     * A functions that will be triggered through a triangulation
```
Here too!

bangerth commented:
I've fixed the two grammar mistakes. It turned out that the next paragraph (in both cases) started with a sentence that was missing the word "that". I fixed that as well.

drwells commented:
Its just a doc fix so lets merge without waiting on the tester.
